ocean_lib/example_config.py

```diff
@@ -5,78 +5,120 @@
+    chain_id = w3.eth.chain_id
+
+    if chain_id not in CONFIG_NETWORK_HELPER:
+        raise ValueError("The chain id for the specific RPC could not be fetched!")
+
+    config_helper = copy.deepcopy(config_defaults)
+    config_helper[SECTION_ETH_NETWORK].update(
+        {
+            NAME_NETWORK_URL: network_url,
+            NAME_CHAIN_ID: chain_id,
+            NETWORK_NAME: CONFIG_NETWORK_HELPER[chain_id][NETWORK_NAME],
+            NAME_STORE_INTERVAL: CONFIG_NETWORK_HELPER[chain_id][NAME_STORE_INTERVAL],
+        }
+    )
+    config_helper[SECTION_RESOURCES].update(
+        {
+            NAME_PROVIDER_URL: CONFIG_NETWORK_HELPER[chain_id][NAME_PROVIDER_URL],
+            NAME_METADATA_CACHE_URI: CONFIG_NETWORK_HELPER[chain_id][
+                NAME_METADATA_CACHE_URI
+            ],
         }
+    )
+    return config_helper
 
-    @staticmethod
-    def get_network_config(network_name: str) -> Dict[str, Dict[str, str]]:
-        """
-        :return: dict
-        """
-        config = ExampleConfig.get_base_config()
-        config["eth-network"]["network"] = get_infura_url(get_infura_id(), network_name)
-        return config
 
+@enforce_types
+class ExampleConfig:
     @staticmethod
-    def _get_config(
-        local_node: bool = True, net_name: Optional[str] = None
-    ) -> Dict[str, Dict[str, str]]:
-        """
-        :return: dict
+    def get_config() -> Config:
+        """Return `Config` containing default values for a given network.
+        Chain ID is determined by querying the RPC specified by `OCEAN_NETWORK_URL` envvar.
         """
-        if local_node:
-            return ExampleConfig.get_base_config()
 
-        return ExampleConfig.get_network_config(net_name)
+        network_url = os.getenv("OCEAN_NETWORK_URL")
+        assert (
+            network_url is not None
+        ), "Cannot use ocean-lib without a specified network URL."
 
-    @staticmethod
-    def get_config_dict(
-        network_name: Optional[str] = None,
-    ) -> Dict[str, Dict[str, str]]:
-        """
-        :return: dict
-        """
-        test_net = network_name or ExampleConfig.get_config_net()
-        local_node = not test_net or test_net in {"local", "ganache"}
-        config_dict = ExampleConfig._get_config(local_node, test_net)
-        logging.debug(
-            f"Configuration loaded for environment `{network_name}`: {config_dict}"
-        )
-        return config_dict
+        assert network_url.startswith("http") or network_url.startswith(
+            "wss"
+        ), "Invalid schema for OCEAN_NETWORK_URL!"
 
-    @staticmethod
-    def get_config(network_name: Optional[str] = None) -> Config:
-        """
-        :return: `Config` instance
-        """
-        return Config(options_dict=ExampleConfig.get_config_dict(network_name))
+        w3 = Web3(get_web3_connection_provider(network_url))
+        chain_id = w3.eth.chain_id
+
+        if chain_id not in CONFIG_NETWORK_HELPER:
```

Comment:
This validation check is performed by `get_config_helper_network` so this line is a duplicate. Please remove it.